### ISSUE TITLE
Erroneous "Discovery only supported for block devices, don't know what to do"

### ISSUE THREAD

**Issue by HaleTom** [systemd/systemd#6756]
### Submission type

 - Bug report

### systemd version the issue has been seen with
```
$ /usr/lib/systemd/systemd --version
systemd 233
+PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 -SECCOMP +BLKID +ELFUTILS +KMOD +IDN default-hierarchy=hybrid
```

### Used distribution

> Manjaro Linux

### In case of bug report: Expected behaviour you didn't see

Something like null output or `filesystem is busy`

### In case of bug report: Unexpected behaviour you saw

```
$ grep backup /proc/mounts
systemd-1 /media/backup autofs rw,relatime,fd=27,pgrp=1,timeout=30,minproto=5,maxproto=5,direct 0 0
/dev/mapper/2TB-backup /media/backup btrfs rw,lazytime,nosuid,nodev,noexec,relatime,compress=zlib,space_cache,subvolid=5,subvol=/ 0 0
$ systemd-mount --umount /media/backup/
Discovery only supported for block devices, don't know what to do.
$ lsblk -f /dev/mapper/2TB-backup
NAME       FSTYPE LABEL      UUID                                 MOUNTPOINT
2TB-backup btrfs  2TB-backup f9530f0e-ed0d-4f76-a58a-e91c737b499f /media/backup
$
```

The WHERE is a mount of a block device, so this is strange to me.

### In case of bug report: Steps to reproduce the problem

See transcript above

**Comment by poettering** [comment]
Pretty sure this got fixed by #6096 already. Please update to a newer systemd. If you manage to reproduce the issue there, please reopen.

**Comment by HaleTom** [comment]
Confirming this was fixed in v234.
